magik-debug-adapter/src/main/java/nl/ramsolutions/sw/magik/debugadapter/MagikDebugAdapter.java

```diff
@@ -228,6 +228,7 @@ public CompletableFuture<Void> disconnect(final DisconnectArguments args) {
           this.threadManager = null;
           this.variableManager = null;
           this.breakpointManager = null;
+          System.exit(1);
```

Comment:
Can you explain why this is needed?

Reply:
When I disconnected from the debug adapter, the Java process would not stop running without this. Maybe this is also a side effect from the change where I am connecting via a socket instead of the jdwp-agent. I will try if the jdwp-agent resolves the issue and remove it then.

Reply:
Also, why exit with a non-zero status code? This indicates the process exits with an error. As far as I can see from the code, the disconnect request from the client is handled properly and thus should not result in an error status code.

This is what the LSP4J docs say about the disconnect call:
>The 'disconnect' request asks the debug adapter to disconnect from the debuggee
> (thus ending the debug session) and then to shut down itself (the debug adapter).
>
> In addition, the debug adapter must terminate the debuggee if it was started with
> the 'launch' request. If an 'attach' request was used to connect to the debuggee,
> then the debug adapter must not terminate the debuggee.
>
> This implicit behavior of when to terminate the debuggee can be overridden with
> the 'terminateDebuggee' argument (which is only supported by a debug adapter
> if the corresponding capability {@link Capabilities#getSupportTerminateDebuggee} is true).